pkg/instancetype/preference/webhooks/admitter_test.go

```diff
@@ -21,14 +20,185 @@ import (
+	obj := &instancetypev1beta1.VirtualMachinePreference{
+		Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
+			PreferSpreadSocketToCoreRatio: socketToCoreRatio,
+			CPU: &instancetypev1beta1.CPUPreferences{
+				PreferredCPUTopology: preferredCPUTopology,
+				SpreadOptions: &instancetypev1beta1.SpreadOptions{
+					Across: spreadAcross,
+					Ratio:  ratio,
+				},
+			},
+		},
+	}
+	return obj
+}
+
+func (h *virtualMachinePreferenceHelper) createObjectForWarning(deprecatedTopology instancetypev1beta1.PreferredCPUTopology) interface{} {
+	return &instancetypev1beta1.VirtualMachinePreference{
+		Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
+			CPU: &instancetypev1beta1.CPUPreferences{
+				PreferredCPUTopology: pointer.P(deprecatedTopology),
+			},
+		},
+	}
+}
+
+func (h *virtualMachinePreferenceHelper) createAdmissionReview(obj interface{}, version string) *admissionv1.AdmissionReview {
+	return createPreferenceAdmissionReview(obj.(*instancetypev1beta1.VirtualMachinePreference), version)
+}
+
+func (h *virtualMachinePreferenceHelper) getAdmitter() interface{} {
+	return h.admitter
+}
+
+func (h *virtualMachinePreferenceHelper) getSpecPath() *k8sfield.Path {
+	return k8sfield.NewPath("spec")
+}
+
+type virtualMachineClusterPreferenceHelper struct {
+	admitter *webhooks.ClusterPreferenceAdmitter
+}
+
+func (h *virtualMachineClusterPreferenceHelper) createObject(preferredCPUTopology *instancetypev1beta1.PreferredCPUTopology, spreadAcross *instancetypev1beta1.SpreadAcross, ratio *uint32, socketToCoreRatio uint32) interface{} {
+	obj := &instancetypev1beta1.VirtualMachineClusterPreference{
+		Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
+			PreferSpreadSocketToCoreRatio: socketToCoreRatio,
+			CPU: &instancetypev1beta1.CPUPreferences{
+				PreferredCPUTopology: preferredCPUTopology,
+				SpreadOptions: &instancetypev1beta1.SpreadOptions{
+					Across: spreadAcross,
+					Ratio:  ratio,
+				},
+			},
+		},
+	}
+	return obj
+}
+
+func (h *virtualMachineClusterPreferenceHelper) createObjectForWarning(deprecatedTopology instancetypev1beta1.PreferredCPUTopology) interface{} {
+	return &instancetypev1beta1.VirtualMachineClusterPreference{
+		Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
+			CPU: &instancetypev1beta1.CPUPreferences{
+				PreferredCPUTopology: pointer.P(deprecatedTopology),
+			},
+		},
+	}
+}
+
+func (h *virtualMachineClusterPreferenceHelper) createAdmissionReview(obj interface{}, version string) *admissionv1.AdmissionReview {
+	return createClusterPreferenceAdmissionReview(obj.(*instancetypev1beta1.VirtualMachineClusterPreference), version)
+}
+
+func (h *virtualMachineClusterPreferenceHelper) getAdmitter() interface{} {
+	return h.admitter
+}
+
+func (h *virtualMachineClusterPreferenceHelper) getSpecPath() *k8sfield.Path {
+	return k8sfield.NewPath("spec")
+}
+
+func runUnsupportedSpreadOptionsTest(helper preferenceTestHelper, preferredCPUTopology instancetypev1beta1.PreferredCPUTopology) {
```

Comment:
You need to fold these tests back within their associated container or Ginkgo will panic.